sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/spanner/changestreams/restriction/TimestampRangeTracker.java

```diff
@@ -45,12 +49,20 @@
 public class TimestampRangeTracker extends RestrictionTracker<TimestampRange, Timestamp>
     implements HasProgress {
 
+  private static final Logger LOG = LoggerFactory.getLogger(TimestampRangeTracker.class);
   protected TimestampRange range;
   protected @Nullable Timestamp lastAttemptedPosition;
   protected @Nullable Timestamp lastClaimedPosition;
+  protected TimeFunction timeFunction;
```

Comment:
You can use Java Clock instead: https://docs.oracle.com/javase/8/docs/api/java/time/Clock.html

Reply:
Oh never mind, you can not, because we need a google cloud Timestamp